pybamm/models/submodels/interface/sei/base_sei.py

```diff
@@ -212,21 +251,32 @@ def _get_standard_concentration_variables(self, variables):
                     f"Loss of capacity to {domain} {reaction_name}[A.h]": Q_sei
                     * self.param.F
                     / 3600,
-                }
-            )
+                    }
+                )
         # Concentration variables are handled slightly differently for SEI on cracks
         elif self.reaction == "SEI on cracks":
-            L_inner_cr = variables[f"{Domain} inner {reaction_name}thickness [m]"]
-            L_outer_cr = variables[f"{Domain} outer {reaction_name}thickness [m]"]
-            roughness = variables[f"{Domain} electrode roughness ratio"]
+            # Case for single layer SEI
+            if self.single_layer_sei:
+                L_sei_cr = variables[f"{Domain} outer {reaction_name}thickness [m]"]
+                roughness = variables[f"{Domain} electrode roughness ratio"]
+
+                n_SEI_cr = L_sei_cr * L_to_n_outer * (roughness - 1)
+
+            # Case for two SEI layers
+            else:
+                L_inner_cr = variables[f"{Domain} inner {reaction_name}thickness [m]"]
+                L_outer_cr = variables[f"{Domain} outer {reaction_name}thickness [m]"]
+                roughness = variables[f"{Domain} electrode roughness ratio"]
 
-            n_inner_cr = L_inner_cr * L_to_n_inner * (roughness - 1)
-            n_outer_cr = L_outer_cr * L_to_n_outer * (roughness - 1)
+                n_inner_cr = L_inner_cr * L_to_n_inner * (roughness - 1)
+                n_outer_cr = L_outer_cr * L_to_n_outer * (roughness - 1)
 
-            n_inner_cr_av = pybamm.x_average(n_inner_cr)
-            n_outer_cr_av = pybamm.x_average(n_outer_cr)
+                n_inner_cr_av = pybamm.x_average(n_inner_cr)
+                n_outer_cr_av = pybamm.x_average(n_outer_cr)
 
-            n_SEI_cr = n_inner_cr + n_outer_cr  # SEI on cracks concentration
+                n_SEI_cr = n_inner_cr + n_outer_cr  # SEI on cracks concentration
```

Comment:
Same comment as before, see if you can reuse some of outer layer work to simplify these blocks.